### PR TITLE
refactor(material/progress-bar): change mat-progress-bar to use MDC's…

### PR DESCRIPTION
**Pull request by wagnermaciel**
… token API

### CODE REVIEW

File: src/material/core/tokens/m2/mdc/_linear-progress.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// `null` indicates that we are intentionally choosing not to emit a slot or value for the token in
+// our CSS.
+@function get-unthemable-tokens() {
+  @return (track-height: 4px);
+}
+
+// Tokens that can be configured through Angular Material's color theming API.
+@function get-color-tokens($config) {
+  $primary: theming.get-color-from-palette(map.get($config, primary));
+  @debug ($primary);
```
Comment (mmalerba) [inline src/material/core/tokens/m2/mdc/_linear-progress.scss:23]: I see @debug and some commented out stuff below

**Review by mmalerba — COMMENTED** [review]
I don't see _progress-bar-theme.scss in this PR

**Comment by wagnermaciel** [comment]
@mmalerba I wasn't able to get the svg background-image to inherit the fill color even after setting it `currentColor`. Because of this, I went with a similar solution to our old implementation, which involves calling `theme-styles` in `_progress-bar-theme.scss`. I know this is not great, so I added to the comment to briefly mention why we cannot work around this by creating our own token slot

https://github.com/angular/components/blob/f2fbd9e0d73b6a2e7b1112e1247dfa75975e3fc9/src/material/progress-bar/_progress-bar-theme.scss#L12-L13

**Review by mmalerba — APPROVED** [review]
LGTM

**Pull request merged**